recipe/cmake-tests/CMakeLists.txt

```diff
@@ -0,0 +1,77 @@
+cmake_minimum_required(VERSION 3.17)
```

Comment:
Would suggest bumping this to 3.18 as some of the buggy behavior we're adding workarounds for didn't exist until 3.18.